src/typescript/frontend/src/components/trade/OrderEntry/LimitOrderEntry.tsx

```diff
@@ -60,6 +64,76 @@ export const LimitOrderEntry: React.FC<{
     };
     const orderSide = orderSideMap[side];
 
+    // validation
+    const _rawValueSize = toRawCoinAmount(
+      values.size ?? 0,
+      marketData?.base?.decimals ?? 0,
+    );
+    const rawValueSize = canBeBigInt(_rawValueSize) // if after conversion is still a decimal, then it's too small anyways
+      ? BigInt(_rawValueSize)
+      : BigInt(0);
+    const rawBaseBalance = BigInt(
+      toRawCoinAmount(
+        baseBalance.data ?? 0, // assume if null, user has 0
+        marketData?.base?.decimals ?? 0, // is this fine?
+      ),
+    );
+    const _rawValuePrice = BigInt(
+      toRawCoinAmount(values.price ?? 0, marketData?.quote?.decimals ?? 0),
+    );
+    const rawValuePrice = canBeBigInt(_rawValuePrice) // if after conversion is still a decimal, then it's too small anyways
+      ? BigInt(_rawValuePrice)
+      : BigInt(0);
+
+    const rawQuoteBalance = BigInt(
+      toRawCoinAmount(
+        quoteBalance.data ?? 0, // assume if null, user has 0
+        marketData?.quote?.decimals ?? 0, // is this fine?
+      ),
+    );
+
+    // validate tick size
+    if (rawValuePrice % BigInt(marketData.tick_size) != BigInt(0)) {
+      setError("price", { message: "INVALID TICK SIZE" });
+      return;
+    }
+
+    // validate Lot size
+    if (rawValueSize % BigInt(marketData.lot_size) != BigInt(0)) {
+      setError("size", { message: "INVALID LOT SIZE" });
+      return;
+    }
+    // validate min size
+    if (rawValueSize < BigInt(marketData.min_size)) {
+      setError("size", { message: "INVALID MIN SIZE" });
+      return;
+    }
+
+    // limit buy -- make sure user has enough quote balance
+    if (orderSide === "bid") {
+      const isValid =
+        BigInt(
+          toRawCoinAmount(
+            rawQuoteBalance.toString(),
+            marketData?.base?.decimals ?? 0,
+          ),
+        ) >=
+        BigInt(rawValueSize) * BigInt(rawValuePrice); // how to get price? esp if market order
+      if (!isValid) {
+        setError("price", { message: "INSUFFICIENT QUOTE BALANCE" });
```

Comment:
I prefer showing insufficient balance errors on size

Reply:
just change to say `INSUFFICIENT BALANCE`?

Reply:
yeah I guess so